lib/ModelAE.py

```diff
@@ -3,28 +3,36 @@
 import time
 import numpy
 from lib.training_data import TrainingDataGenerator, stack_images
+from keras.losses import *
 
-encoderH5 = 'encoder.h5'
-decoder_AH5 = 'decoder_A.h5'
-decoder_BH5 = 'decoder_B.h5'
 
 class ModelAE:
-    def __init__(self, model_dir):
+    def __init__(self, model_dir, filename_modifier):
+
         self.model_dir = model_dir
+        self.encoder_filename = self.model_dir + '/encoder_' + filename_modifier + '.h5'
+        self.decoder_A_filename = self.model_dir + '/decoder_A_' + filename_modifier + '.h5'
+        self.decoder_B_filename = self.model_dir + '/decoder_B_' + filename_modifier + '.h5'
 
         self.encoder = self.Encoder()
         self.decoder_A = self.Decoder()
         self.decoder_B = self.Decoder()
-
+        self.loss_function = mean_absolute_error
         self.initModel()
+        
+    #
+    # set_loss_function has to be called *before* the model is initialized!
+    #
+    def set_loss_function(self, loss_function):

```

Comment:
whats the reasoning behind this function over passing the loss function to __init__?